### PR TITLE
Implement t.navigateTo (part of #240)

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin @VasilyStrelyaev

### CODE REVIEW

File: src/api/test-controller.js [hunk 1]
```diff
@@ -16,8 +16,12 @@ import {
     SelectTextAreaContentCommand,
     SelectEditableContentCommand,
     PressKeyCommand,
+<<<<<<< bf5c275f1e11cee91bf4413c7344aec6619d424a
     WaitCommand,
     WaitForElementCommand
+=======
+    NavigateToCommand
+>>>>>>> Implement t.navigateTo (part of #240)
```
Comment (VasilyStrelyaev) [inline src/api/test-controller.js]: nice!

File: test/functional/fixtures/api/es-next/navigate-to/test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+var expect = require('chai').expect;
+
+
+describe('[API] t.navigateTo', function () {
+    it('Should navigate to the specified url', function () {
+        return runTests('./testcafe-fixtures/navigate-to-test.js', 'Navigate to another page');
+    });
+
+    it('Should validate url argument', function () {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/api/es-next/navigate-to/test.js]:
```
    it('Should validate the url argument', function () {
```

**Comment by VasilyStrelyaev** [comment]
\r-

File: test/functional/fixtures/api/es-next/navigate-to/test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+describe('[API] t.navigateTo', function () {
+    it('Should navigate to the specified url', function () {
+        return runTests('./testcafe-fixtures/navigate-to-test.js', 'Navigate to another page');
+    });
+
+    it('Should validate the url argument', function () {
+        return runTests('./testcafe-fixtures/navigate-to-test.js', 'Incorrect protocol', { shouldFail: 'true' })
+            .catch(function (errs) {
+                expect(errs[0]).contains('The url argument specifies a URL that uses an unsupported ftp protocol. Only HTTP and HTTPS are supported, as well as protocol-relative and relative URLs.');
```
Comment (inikulin) [inline test/functional/fixtures/api/es-next/navigate-to/test.js]: Too bad we don't have `://` after protocol name in error message. Can we add this functionality in context of this PR?

**Comment by AndreyBelym** [comment]
FPR

File: src/errors/test-run/templates.js [hunk 4]
```diff
@@ -87,7 +87,7 @@ export default {
     `),
 
     [TYPE.actionUnsupportedUrlProtocolError]: err => markup(err, `
-        The <code>${err.argumentName}</code> argument specifies a URL that uses an unsupported <code>${err.protocol}</code> protocol. Only HTTP and HTTPS are supported, as well as protocol-relative and relative URLs.
+        The <code>${err.argumentName}</code> argument specifies a URL that uses an unsupported <code>${err.protocol}</code>:// protocol. Only HTTP:// and HTTPS:// are supported, as well as protocol-relative and relative URLs.
```
Comment (VasilyStrelyaev) [inline src/errors/test-run/templates.js]: ok, I agree that there are protocols like `mail` or application protocols that may need `://` in the error message to make it clear what we are talking about. But that's hardly the case with HTTP and HTTPS...
```
        The <code>${err.argumentName}</code> argument specifies a URL that uses an unsupported '<code>${err.protocol}</code>://' protocol. Only HTTP and HTTPS are supported, as well as protocol-relative and relative URLs.
```

**Comment by VasilyStrelyaev** [comment]
\r-

File: src/errors/test-run/templates.js [hunk 5]
```diff
@@ -87,7 +87,7 @@ export default {
     `),
 
     [TYPE.actionUnsupportedUrlProtocolError]: err => markup(err, `
-        The <code>${err.argumentName}</code> argument specifies a URL that uses an unsupported <code>${err.protocol}</code> protocol. Only HTTP and HTTPS are supported, as well as protocol-relative and relative URLs.
+        The <code>${err.argumentName}</code> argument specifies a URL that uses an unsupported <code>${err.protocol}</code>:// protocol. Only HTTP and HTTPS are supported, as well as protocol-relative and relative URLs.
```
Comment (inikulin) [inline src/errors/test-run/templates.js]: `://` should be in `<code>` content as well

**Comment by inikulin** [comment]
\r-

**Comment by AlexanderMoskovkin** [comment]
Hmm. something wrong with the tests. @AndreyBelym please check it locally. Maybe it relates to some modules updating (for example `testcafe-hammerhead@8.2.0` was released

**Comment by AndreyBelym** [comment]
Okay, I'll take a look.

**Comment by AlexanderMoskovkin** [comment]
It seems modules versions were not changed since the last successful running. Take a look at the screencast maybe we can see something wrong there

**Comment by AndreyBelym** [comment]
Have found on [screencast](https://travis-ci.org/AndreyBelym/testcafe/jobs/130527378), that tests failures was caused by Chrome hanging (e.g. at 11:06, 13:36). 

Interesting, that this commit has passed functional tests on my travis account: https://travis-ci.org/AndreyBelym/testcafe/jobs/130527378

**Comment by AndreyBelym** [comment]
FPR. Unable to reproduce previous problem; I'm totally sure it was not caused by this PR. Most likely it was a Saucelabs overload, because one time it was unable to start a VM with IE11([link](https://saucelabs.com/tests/4ebfb3bbdc7c46cb95fc8888ba095bde)). However we need to pay attention to the strange timeouts in next functional test runs, to be 200% sure that it was not our issue.

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Pull request merged**